cli/src/main/java/com/devonfw/tools/ide/git/GitContextImpl.java

```diff
@@ -51,12 +52,38 @@ public boolean fetchIfNeeded(Path repository, String remote, String branch) {
   public boolean isRepositoryUpdateAvailable(Path repository) {
 
     verifyGitInstalled();
-    String localCommitId = runGitCommandAndGetSingleOutput("Failed to get the local commit id.", repository, "rev-parse", "HEAD");
-    String remoteCommitId = runGitCommandAndGetSingleOutput("Failed to get the remote commit id.", repository, "rev-parse", "@{u}");
-    if ((localCommitId == null) || (remoteCommitId == null)) {
+    Path commitIdFile = this.context.getIdeHome().resolve(IdeContext.SETTINGS_COMMIT_ID);
+    if (!Files.exists(commitIdFile)) {
+      saveCurrentCommitId(repository);
+    }
+    String trackedCommitId;
+    try {
+      trackedCommitId = Files.readString(commitIdFile);
+    } catch (IOException e) {
       return false;
     }
-    return !localCommitId.equals(remoteCommitId);
+
+    String remoteCommitId = runGitCommandAndGetSingleOutput("Failed to get the remote commit id.", repository, "rev-parse", "@{u}");
+    return !trackedCommitId.equals(remoteCommitId);
+  }
+
+  @Override
+  public void saveCurrentCommitId(Path repository) {
+
+    if (!Files.exists(repository.resolve(GitContext.GIT_FOLDER))) {
+      try {
+        repository = repository.toRealPath();
+      } catch (IOException e) {
+        throw new IllegalStateException("Couldn't resolve settings to real path.", e);
+      }
```

Comment:
you have duplicated this code fragment in multiple places. I have made some suggestions to improve this.
This already indicates that this should go to a central place and should be reused.
IMHO this feature is something special for our `settings` and I would recommend to keep the logic out of `GitContext`.
You could add a `getSettingsGitRepository()` method to `IdeContext` that currently would just delegate to `getSettingsPath()` but could then be extended by you to do this extra logic to find the real git repo with the `.git` folder.
Also you can then detect if no `.git` folder was found at all to resolve #826.
IMHO you should then log an error but return `null`.
In the use-case of #826 we just wanted to check for an update of the settings. If that is not possible, we could still continue regular commands. IMHO commands like `--version` or `--help` should not trigger online network activity after all and especially not fail because of that.

Reply:
forgot to mark this as done too